src/ptyfwd/ptyfwd-tool.c

```diff
@@ -208,14 +210,22 @@ static int run(int argc, char *argv[]) {
 
         peer_fd = safe_close(peer_fd);
 
-        r = event_add_child_pidref(event, &exit_source, &pidref, WEXITED, helper_on_exit, NULL);
+        r = event_add_child_pidref(event, &child, &pidref, WEXITED, helper_on_exit, NULL);
         if (r < 0)
                 return log_error_errno(r, "Failed to add child event source: %m");
 
-        r = sd_event_source_set_child_process_own(exit_source, true);
+        r = sd_event_source_set_child_process_own(child, true);
         if (r < 0)
                 return log_error_errno(r, "Failed to take ownership of child process: %m");
 
+        r = sd_event_source_set_priority(child, SD_EVENT_PRIORITY_IMPORTANT);
```

Comment:
Maybe better to add a short comment why we increase the priority for the child event.